File: src/main/java/net/casual/arcade/ducks/MutableAdvancements.java
package net.casual.arcade.ducks;

import net.minecraft.advancements.AdvancementHolder;

import java.util.Collection;

public interface MutableAdvancements {
	void arcade$addAllAdvancements(Collection<AdvancementHolder> advancements);

	void arcade$addAdvancement(AdvancementHolder advancement);

	void arcade$removeAdvancement(AdvancementHolder advancement);
}

File: src/main/kotlin/net/casual/arcade/utils/ducks/MutableAdvancements.kt
package net.casual.arcade.utils.ducks

import net.minecraft.advancements.AdvancementHolder

internal interface MutableAdvancements {
    fun addAllAdvancements(advancements: Collection<AdvancementHolder>)

    fun addAdvancement(advancement: AdvancementHolder)

    fun removeAdvancement(advancement: AdvancementHolder)
}